adam/learner/integrated_learner.py

```diff
@@ -217,92 +270,116 @@ def _linguistic_descriptions_from_semantics(
     def _add_determiners(
         self, object_node: ObjectSemanticNode, cur_string: Tuple[str, ...]
     ) -> Tuple[str, ...]:
+        # handle Chinese Classifiers by casing on the words -- this is hackish
         if (
             self.object_learner._language_mode  # pylint: disable=protected-access
-            != LanguageMode.ENGLISH
+            == LanguageMode.CHINESE
         ):
-            return cur_string
-        # If plural, we want to strip any "a" that might preceed a noun after "many" or "two"
-        if isinstance(self.attribute_learner, SubsetPluralLearnerNew):
-            if "a" in cur_string:
-                a_position = cur_string.index("a")
-                if a_position > 0 and cur_string[a_position - 1] in ["many", "two"]:
+            if self.plural_learner:
+                return tuple([token for token in cur_string if token[:3] != "yi1"])
+            # specially handle the case of my and your in Chinese since these organize the classifier and attribute differently
+            if cur_string[0] in ["ni3 de", "wo3 de"] and len(cur_string) > 1:
+                my_your_classifier = get_classifier_for_string(cur_string[1])
+                if my_your_classifier:
                     return tuple(
-                        [token for i, token in enumerate(cur_string) if i != a_position]
+                        chain((cur_string[0], my_your_classifier), cur_string[1:])
                     )
-        # English-specific hack to deal with us not understanding determiners:
-        # https://github.com/isi-vista/adam/issues/498
-        # The "is lower" check is a hack to block adding a determiner to proper names.
-        # Ground is a specific thing so we special case this to be assigned
-        if object_node.concept == GROUND_OBJECT_CONCEPT:
-            return tuple(chain(("the",), cur_string))
-        elif (
-            object_node.concept.debug_string not in MASS_NOUNS
-            and object_node.concept.debug_string.islower()
-            and not cur_string[0] in ENGLISH_BLOCK_DETERMINERS
-        ):
-            return tuple(chain(("a",), cur_string))
+                else:
+                    return cur_string
+            # get the classifier and add it to the language
+            classifier = get_classifier_for_string(cur_string[-1])
+            if classifier:
+                return tuple(chain((classifier,), cur_string))
+            else:
+                return cur_string
+
+        # handle English determiners
         else:
```

Comment:
Should this instead be an `elif` to check if the language mode is English? I realize we only have two languages at the moment but handling it this way means if someone doesn't fix this specific hack all languages will get this processing when that may not be desired?

Reply:
I agree.